backend/compact-connect/lambdas/python/common/cc_common/event_bus_client.py

```diff
@@ -120,6 +120,7 @@ def publish_license_encumbrance_event(
         provider_id: str,
         jurisdiction: str,
         license_type_abbreviation: str,
+        effective_date: str,
```

Comment:
This should be a `datetime`, not a `str`. We generally should aim to serialize objects as late as possible, so we're passing the richer objects around internally.